rs/index/src/ivf/files/invalidated_ids.rs

```diff
@@ -26,6 +29,66 @@ impl InvalidatedIdsStorage {
         }
     }
 
+    pub fn read(base_directory: &str) -> Result<Self> {
+        let base_path = Path::new(base_directory);
+        // Create a storage from scratch if none exists.
+        if !base_path.exists() || !base_path.is_dir() {
+            return Ok(Self::new(base_directory, Self::DEFAULT_BACKING_FILE_SIZE));
+        }
+
+        // Get the list of invalidated ids files
+        let mut invalidated_ids_files: Vec<String> = read_dir(base_directory)?
+            .filter_map(|entry| entry.ok()) // Ignore errors
+            .filter_map(|entry| {
+                let file_name = entry.file_name();
+                let file_name_str = file_name.to_str()?.to_string();
+
+                if file_name_str.starts_with("invalidated_ids.bin.") {
+                    Some(entry.path().to_str()?.to_string())
+                } else {
+                    None
+                }
+            })
+            .collect();
+
+        if invalidated_ids_files.is_empty() {
+            return Ok(Self::new(base_directory, Self::DEFAULT_BACKING_FILE_SIZE));
+        }
+
+        // Sort the list in lexicographical order so the last file's size gives us
+        // the current offset
+        invalidated_ids_files.sort();
+        let current_offset = metadata(invalidated_ids_files.last().unwrap())?.len() as usize;
```

Comment:
`invalidated_ids.bin.2` will go after `invalidated_ids.bin.11`, so this might be a bug

Reply:
good catch, fixed